Fix incorrect NTP server settings behavior

brandboat opened the pull request:
**Problem:**
Harvester NTP server settings not correctly mapping to NodeConfig ntpConfigs.
https://github.com/harvester/harvester/blob/226f72aa58532071232ad548e27970c1cec55733/pkg/controller/master/setting/ntp_server.go#L54 retrieve ntp settings from NodeConfig and https://github.com/harvester/harvester/blob/226f72aa58532071232ad548e27970c1cec55733/pkg/util/ntp.go#L19-L21 append the ntp server if it not exist in the current net servers slice. This causes an issue where even if we clear the NTP server settings and add new ones, NodeConfig will append the new NTP servers to the existing configuration instead of overriding the original settings.

**Solution:**
Reflect what user set `ntp-servers` in settings CR to NodeConfig CR `ntpConfigs`

This pr also addressed the follows:
- Add validation rule to check if there is any duplicate NTP servers in settings CR
- Enhance error message when setup ntp server string start with https
- Add validateNTPServers unit tests

**Related Issue:**
https://github.com/harvester/harvester/issues/5876

**Test plan:**
Prerequisites:
- Set up a Harvester cluster

Test steps:
- Dashboard -> Settings -> NTP Servers, remove `0.suse.pool.ntp.org` `1.suse.pool.ntp.org` and add `time1.google.com` and `time2.google.com`
- Wait a couple seconds, go to each Harvester node and check `cat /etc/systemd/timesyncd.conf`. The content should be `time1.google.com` and `time2.google.com`
- Run `systemctl status systemd-timesyncd.service` and check service status is `active` and time server is sync from 216.239.35.0:123 (time1.google.com)
```
● systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2024-05-27 09:41:14 UTC; 20s ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 4053 (systemd-timesyn)
     Status: "Initial synchronization to time server 216.239.35.0:123 (time1.google.com)."
      Tasks: 2
     CGroup: /system.slice/systemd-timesyncd.service
             └─ 4053 /usr/lib/systemd/systemd-timesyncd

May 27 09:41:14 harvester-node-0 systemd[1]: Starting Network Time Synchronization...
May 27 09:41:14 harvester-node-0 systemd[1]: Started Network Time Synchronization.
May 27 09:41:15 harvester-node-0 systemd-timesyncd[4053]: Initial synchronization to time server 216.239.35.0:123 (time1.google.com).
```

bk201 reviewed (commented):
LGTM, but let @Vicente-Cheng check if there was a reason to keep the old values.

brandboat committed:
Update go-common version

Signed-off-by: Cooper Tseng <cooper.tseng@suse.com>

brandboat committed:
Fix incorrect NTP server settings behavior

Harvester NTP server settings not correctly mapping to NodeConfig ntpConfigs
This commit address the follows:
- Reflect what user set in settings CR ntp-servers to NodeConfig CR ntpConfigs
- Add validation rule to check if there is any duplicate NTP servers in settings CR
- Enhance error message when setup ntp server start with https
- Add validateNTPServers unit tests

Signed-off-by: Cooper Tseng <cooper.tseng@suse.com>

Vicente-Cheng reviewed (commented):
Just a nit. Please help to check with that.

BTW, I cannot remember why we just append instead of replace, and it looks like it should be fine with replace. So let fix the old mechanism!

Vicente-Cheng reviewed (approved):
LGTM, thanks.
